Add support for managing a factory's denied devices

doanac opened the pull request.

vkhoroz commented on subcommands/devices/list_denied.go:
```diff
@@ -0,0 +1,76 @@
+)
+
+func init() {
+	paginationLimits = []int{10, 20, 30, 40, 50, 100, 200, 500, 1000}
```
I think it's worth extracting this as a constant... it's the same for both device list and denied device list.
Maybe, even extract into common (if can be done easily, not sure about this).

vkhoroz commented on subcommands/devices/list_denied.go:
```diff
@@ -0,0 +1,76 @@
+		found := false
+		for i := 0; i < len(os.Args); i++ {
+			arg := os.Args[i]
+			if len(arg) > 2 && arg[:2] == "-p" {
```
IIRC `-p <page>` is as valid as `-p<page>`.

So, we need to also check for `if len(arg) == 2 && arg == "-p"` and use the same logic as below for a `--page`.

vkhoroz commented on subcommands/devices/list_denied.go:
```diff
@@ -0,0 +1,76 @@
+			if len(arg) > 2 && arg[:2] == "-p" {
+				fmt.Printf("-p%d ", showPage+1)
+				found = true
+			} else if len(arg) == 6 && arg[:6] == "--page" {
```
This can come in two forms: `--page <page>` and `--page=<page>`.
Need to handle both.

vkhoroz commented on subcommands/devices/list_denied.go:
```diff
@@ -0,0 +1,76 @@
+			fmt.Print("-p", showPage+1)
+		}
+		fmt.Println()
+	}
```
I don't see an easier way to write this logic.
But I have a question if we need it at all.
A contemporary user should be able to understand and add the `-p` argument.

doanac replied:
I just copied what you did in device-list for this. I guess I should first try and make this common code first.

vkhoroz replied:
it wasn't my logic though :-)

vkhoroz commented on subcommands/devices/delete_denied.go:
```diff
@@ -0,0 +1,37 @@
+			os.Exit(1)
+		} else {
+			fmt.Printf("ok\n")
+		}
```
Why not `DieNotNil`?

Also, it is the first command which prints 'ok' in case of success. All other commands like this print nothing.

doanac replied:
i copied the code as-is from delete.go. i think it was done like this because it can remove multiple, so just dying isn't going to show you which device it failed on.

vkhoroz replied:
Maybe, it is an instance of code I missed when refactoring to use DieNotNil everywhere; maybe it was added later; I don't know.
What I know is that `commands.DieNotNil(api.DeviceDelete(factory, name), "failed"); fmt.Println("ok")` is equivalent; but I'd prefer `commands.DieNotNil(api.DeviceDelete(factory, name))` as I see no value in these `failed, ok` things.

Just dying will show clearly on which device it failed due to these `Deleting xyz...` messages.

doanac commented:
@vkhoroz - i've rebased this on a change that first uses common device list display logic. Then I've fixed the pagination display logic with [3aa5e3f](https://github.com/foundriesio/fioctl/pull/150/commits/3aa5e3f5e484d24135ff7de51b31c7bbfc2a4fce). I know you aren't a big fan of that, but I think we should try and help there cause its not very obvious to me what to do without it.

MrCry0 commented on subcommands/devices/list.go:
```diff
@@ -183,10 +183,17 @@ func showDeviceList(dl *client.DeviceList, showColumns []string) {
 				fmt.Printf("-p%d ", showPage+1)
 				found = true
-			} else if len(arg) == 6 && arg[:6] == "--page" {
+			} else if arg == "-p" {
```
The following 2 blocks may be rewritten as
```
} else if arg == "-p" || arg == "--page" {
	fmt.Printf("-p%d ", showPage+1)
	found = true
	i++
}
```

vkhoroz commented on subcommands/devices/list.go:
```diff
@@ -183,10 +183,13 @@ func showDeviceList(dl *client.DeviceList, showColumns []string) {
 				i++
+			} else if len(arg) > 7 && arg[:7] == "--page=" {
+				fmt.Printf("-p%d ", showPage+1)
+				found = true
```
I'm sorry for nitpicking, but this can also be combined with the above as:
`if (len(arg) > 2 && arg[:2] == "-p") || (len(arg) > 7 && arg[:7] == "--page=") {...}`

doanac replied:
not a nit. it cleans things up. force push fixed

vkhoroz reviewed (approved):
Nice, LGTM.

vkhoroz commented:
ship it!

MrCry0 reviewed (approved):
LGTM
